src/util/createInstantSearchComponent.js

```diff
@@ -0,0 +1,60 @@
+import { createSuitMixin } from '../mixins/suit';
+import { version } from '../../package.json'; // rollup does pick only what needed from json
+import { _objectSpread } from './polyfills';
+
+export const createInstantSearchComponent = component =>
+  _objectSpread(
+    {
+      mixins: [createSuitMixin({ name: 'InstantSearch' })],
+      provide() {
+        return {
+          instantSearchInstance: this.instantSearchInstance,
+        };
+      },
+      watch: {
+        searchClient(searchClient) {
+          this.instantSearchInstance.helper.setClient(searchClient).search();
+        },
+        indexName(indexName) {
+          this.instantSearchInstance.helper.setIndex(indexName).search();
+        },
+        stalledSearchDelay(stalledSearchDelay) {
+          // private InstantSearch.js API:
+          this.instantSearchInstance._stalledSearchDelay = stalledSearchDelay;
+        },
+        routing() {
+          throw new Error(
+            'routing configuration can not be changed dynamically at this point.' +
```

Comment:
Why is that?

Maybe phrase it:

> Routing is not supported [...]

Reply:
because I couldn't figure out how to monkey patch InstantSearch to take the new Routing configuration

Reply:
(this is not from this PR by the way)